server/manager/src/main/java/org/apache/accumulo/manager/tableOps/Utils.java

```diff
@@ -56,15 +56,29 @@ public class Utils {
   private static final byte[] ZERO_BYTE = {'0'};
   private static final Logger log = LoggerFactory.getLogger(Utils.class);
 
-  public static void checkTableDoesNotExist(ServerContext context, String tableName,
+  public static void checkTableNameDoesNotExist(ServerContext context, String tableName,
       TableId tableId, TableOperation operation) throws AcceptableThriftTableOperationException {
 
-    TableId id = context.getTableNameToIdMap().get(tableName);
-
-    if (id != null && !id.equals(tableId)) {
-      throw new AcceptableThriftTableOperationException(null, tableName, operation,
-          TableOperationExceptionType.EXISTS, null);
+    try {
+      for (String tid : context.getZooReader()
+          .getChildren(context.getZooKeeperRoot() + Constants.ZTABLES)) {
+        String zTablePath = context.getZooKeeperRoot() + Constants.ZTABLES + "/" + tid;
+        byte[] tname = context.getZooReader().getData(zTablePath + Constants.ZTABLE_NAME);
+        if (tname == null) {
+          log.warn("Malformed table entry in ZooKeeper at {}", zTablePath);
+          continue;
+        }
```

Comment:
Could throw an exception for  this if its not expected.

```suggestion
        Preconditions.checkState(tname != null, "Malformed table entry in ZooKeeper at %s", zTablePath);
```

Reply:
@keith-turner  - see changes in 60ccdc6